File: client/src/app/modules/miscellaneous/services/attachment.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { AttachmentType } from '../../miscellaneous/model/AttachmentType';

@Injectable({
  providedIn: 'root'
})
export class AttachmentTypeService extends AppRestService<AttachmentType>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getAttachmentTypes() {
    return this.getListCached(new HttpParams(), "attachment-types");
  }

  addOrUpdateAttachmentType(attachmentType: AttachmentType) {
    this.clearListCache(new HttpParams(), "attachment-types");
    return this.addOrUpdate(new HttpParams(), "attachment-type", attachmentType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/AttachmentTypeService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.AttachmentType;

public interface AttachmentTypeService {
    public List<AttachmentType> getAttachmentTypes();

    public AttachmentType getAttachmentType(Integer id);

    public AttachmentType addOrUpdateAttachmentType(AttachmentType attachmentType);
}
